src/traces/parcoords/attributes.js

```diff
@@ -0,0 +1,188 @@
+            ].join(' ')
+        }
+    },
+
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+        id: {
+            valType: 'string',
+            role: 'info',
+            description: 'Identifier of a dimension. Must be a unique string across all dimensions.'
+        },
+        label: {
+            valType: 'string',
+            role: 'info',
+            description: 'The shown name of the dimension.'
+        },
+        tickvals: axesAttrs.tickvals,
+        ticktext: axesAttrs.ticktext,
+        visible: {
+            valType: 'boolean',
+            dflt: true,
+            role: 'info',
+            description: 'Shows the dimension when set to `true` (the default). Hides the dimension for `false`.'
+        },
+        range: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'The domain range that represents the full, shown axis extent. Defaults to the `values` extent.',
+                'Must be an array of `[fromValue, toValue]` with finite numbers as elements.'
+            ].join(' ')
+        },
+        constraintrange: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'The domain range to which the filter on the dimension is constrained. Must be an array',
+                'of `[fromValue, toValue]` with finite numbers as elements.'
+            ].join(' ')
+        },
+        values: {
+            valType: 'data_array',
+            role: 'info',
+            description: [
+                'Dimension values. `values[n]` represents the value of the `n`th point in the dataset,',
+                'therefore the `values` vector for all dimensions must be the same (longer vectors',
+                'will be truncated). Each value must be a finite number.'
+            ].join(' ')
+        },
+        description: 'The dimensions (variables) of the parallel coordinates chart. 2..63 dimensions are supported.'
+    },
+
+    tickdistance: {
+        valType: 'number',
+        dflt: 50,
+        min: 32,
+        role: 'style',
+        description: 'The desired approximate tick distance (in pixels) between axis ticks on an axis.'
+    },
+
+    padding: {
+        valType: 'number',
+        dflt: 80,
+        min: 0,
+        role: 'style',
+        description: 'The desired space for displaying axis labels and domain ranges around the actual parcoords.'
+    },
+
+    line: extendFlat({},
+
+        colorAttributes('line'),
+
+        {
+            showscale: {
+                valType: 'boolean',
+                role: 'info',
+                dflt: false,
+                description: [
+                    'Has an effect only if `marker.color` is set to a numerical array.',
+                    'Determines whether or not a colorbar is displayed.'
+                ].join(' ')
+            },
+            colorbar: colorbarAttrs
+        },
+
+        {
+            contextopacity: {
```

Comment:
Though not an initial requirement, `parcoords` implementations that can render (tens of) thousands of lines eventually need to support `opacity` e.g. the third figure [here](https://doi.ieeecomputersociety.org/cms/Computer.org/dl/trans/tg/2010/03/figures/ttg20100304399.gif). So I was thinking, let's save the name `opacity` for that use to avoid a more difficult API decision. Maybe there's some other solution that would be more Plotly-like.